### PR TITLE
[BUGFIX] Allow ItemProcessingService as param in execute function of …

### PR DESCRIPTION
**Pull request by tobiasgraeber**
Resolves https://github.com/FluidTYPO3/flux/issues/2143 also see https://github.com/FluidTYPO3/flux/issues/2143#issuecomment-1855795050

### CODE REVIEW

File: Classes/Integration/MultipleItemsProcFunc.php [hunk 1]
```diff
@@ -32,7 +33,7 @@ public static function register(string $table, string $field, ?string $additiona
         $GLOBALS['TCA'][$table]['columns'][$field]['config']['itemsProcFunc'] = $newFunction;
     }
 
-    public function execute(array &$parameters, FormDataProviderInterface $formDataProvider): void
+    public function execute(array &$parameters, ItemProcessingService | FormDataProviderInterface $formDataProvider): void
```
Comment (NamelessCoder) [inline Classes/Integration/MultipleItemsProcFunc.php:36]: Unfortunately I can't accept code with union types - Flux still has to support TYPO3v10 which means supporting PHP 7.4 which sadly doesn't support union types. If this method has to support two different classes as second argument, we have to remove the type and add a PHPDOC instead (which only describes this single argument and notes the two possible classes as type; we don't PHPDOC things that are implied by strict types).

File: Tests/Unit/Integration/MultipleItemsProcFuncTest.php [hunk 2]
```diff
@@ -31,10 +32,12 @@ public function testRegistersFunction(): void
 
     public function testExecutesFunction(): void
     {
-        $provider = $this->getMockBuilder(FormDataProviderInterface::class)->getMockForAbstractClass();
+        $formDataProviderInterface = $this->getMockBuilder(FormDataProviderInterface::class)->getMockForAbstractClass();
```
Comment (NamelessCoder) [inline Tests/Unit/Integration/MultipleItemsProcFuncTest.php:35]: If one of these classes only exists on some versions of TYPO3, then this test method should be split in two and a check made for `class_exists` and if it doesn't, do `$this->markTestSkipped('Skipped on v12');` so the test won't run and always fail on that TYPO3 version.

**Comment by NamelessCoder** [comment]
(preconfigured reply about branching)

Because you created this pull request based on your fork's development branch, once it is merged you will need to either force-push, re-fork or re-create the development branch in your fork. For your own benefit, future pull requests should be created from a branch that you create in your fork with starting point development. That way you won't have conflicts in your fork's development branch; and you can safely create multiple pull requests with one commit each and one branch per pull request.

All FluidTYPO3 repositories use git flow (https://datasift.github.io/gitflow/IntroducingGitFlow.html). Your pull requests should all be created based on feature branches.

**Comment by NamelessCoder** [comment]
Thanks for taking care of this, @tobiasgraeber

**Comment by tobiasgraeber** [comment]
Thank you too! 😊

**Pull request merged**